packages/contractkit/src/wrappers/Governance.ts

```diff
@@ -1,21 +1,103 @@
+import { concurrentMap } from '@celo/utils/lib/async'
+import { zip } from '@celo/utils/lib/collections'
 import BigNumber from 'bignumber.js'
+import { Transaction } from 'web3/eth/types'
+import { Address } from '../base'
 import { Governance } from '../generated/types/Governance'
-import { BaseWrapper, proxyCall, toBigNumber } from './BaseWrapper'
+import {
+  BaseWrapper,
+  bufferToString,
+  fromSolidityBytes,
+  identity,
+  proxyCall,
+  proxySend,
+  stringToBuffer,
+  toSolidityBytes,
+  toTransactionObject,
+  tupleParser,
+  valueToBigNumber,
+  valueToInt,
+  valueToString,
+} from './BaseWrapper'
 
-export interface StageDurations {
-  approval: BigNumber // seconds
-  referendum: BigNumber // seconds
-  execution: BigNumber // seconds
+export enum ProposalStage {
+  None = 'None',
+  Queued = 'Queued',
+  Approval = 'Approval',
+  Referendum = 'Referendum',
+  Execution = 'Execution',
+  Expiration = 'Expiration',
+}
+
+export interface ProposalStageDurations {
+  [ProposalStage.Approval]: BigNumber // seconds
+  [ProposalStage.Referendum]: BigNumber // seconds
+  [ProposalStage.Execution]: BigNumber // seconds
 }
 
 export interface GovernanceConfig {
   concurrentProposals: BigNumber
   dequeueFrequency: BigNumber // seconds
   minDeposit: BigNumber
-  queueExpiry: BigNumber // seconds
-  stageDurations: StageDurations
+  queueExpiry: BigNumber
+  stageDurations: ProposalStageDurations
+}
+
+export interface ProposalMetadata {
+  proposer: Address
+  deposit: BigNumber
+  timestamp: BigNumber
+  transactionCount: number
+}
+
+export type ProposalTransaction = Pick<Transaction, 'to' | 'input' | 'value'>
+export type ProposalParams = Parameters<Governance['methods']['propose']>
+export class Proposal {
```

Comment:
i think it's better to have 
```ts
type Proposal = ProposalTransaction[]
```
and just an utility function `toParams`.. which probably you only need on the wrapper, so it can be just part of it

a Class "feels" to much just to wrap an array